src/pgwire/src/protocol.rs

```diff
@@ -280,8 +281,17 @@ where
 
         self.coord_client.reset_canceled();
 
+        // TODO(guswynn): add spans to all message types. Currently we don't
+        // to avoid useless spans
+        let message_name = message.as_ref().map(|m| m.name()).unwrap_or_default();
+
         let next_state = match message {
-            Some(FrontendMessage::Query { sql }) => self.query(sql).await?,
+            Some(FrontendMessage::Query { sql }) => {
+                let query_root_span =
+                    tracing::debug_span!(parent: None, "advance_ready", otel.name = message_name);
+                OpenTelemetryContext::obtain().attach_as_link_to(&query_root_span);
```

Comment:
Wait if this is the same as `Span::follows_from` it seems less confusing (to me) to use `span.follows_from(Span::current());`?

Reply:
this is much cleaner! updated!! and ill add this advice to the best practices pr!